### ISSUE TITLE
Unable to resolve your shell env **(code null, signal SIGSEGV)

### ISSUE THREAD

**Issue by Ramblestsad** [microsoft/vscode#138708]
Does this issue occur when all extensions are disabled?: Yes/No
yes

- VS Code Version:
- OS Version:

Steps to Reproduce:

1. open vscode
![image](https://user-images.githubusercontent.com/45003009/145352205-fc82daf0-bd52-4743-b2ab-dbea676bc732.png)
![image](https://user-images.githubusercontent.com/45003009/145352230-977d87a4-a098-4a01-9903-f36781e26643.png)

**Comment by Ramblestsad** [comment]
I found that I can type **code** in my Iterm.app to open vscode and everything seems ok.

**Comment by Tyriar** [comment]
2 reports of SIGSEGV on mac universal build when resolving shell environments.

**Comment by gerardkok** [comment]
I have this problem now too after upgrading to Monterey (12.1) from Big Sur (11.6.1) on Intel (still using bash). I didn't change anything else, so perhaps it's Monterey-related.

**Comment by deepak1556** [comment]
Sorry missed this thread, can you give our exploration builds a try which comes a newer runtime version and see if the problem persists. Thanks!

[MacOS Universal Download](https://az764295.vo.msecnd.net/exploration/8a3f490b5f148653b87327b53a26550f695939f5/VSCode-darwin-universal.zip)

**Comment by gerardkok** [comment]
I tried the exploration build you linked, but I'm afraid the problem is still present.

The exploration build I tried:
```
Version: 1.64.0-exploration (Universal)
Commit: 8a3f490b5f148653b87327b53a26550f695939f5
Date: 2022-01-25T10:40:09.616Z
Electron: 16.0.7
Chromium: 96.0.4664.110
Node.js: 16.9.1
V8: 9.6.180.21-electron.0
OS: Darwin x64 21.2.0
```

The `main.log` has lines like the following:
```
[2022-01-26 14:53:22.982] [main] [info] Starting extension host with pid 38016 (fork() took 4 ms).
[2022-01-26 14:53:22.983] [main] [info] ExtensionHostStarterWorker.start() took 11 ms.
[2022-01-26 14:53:23.237] [main] [error] RequestService#request resolving shell environment failed {}
[2022-01-26 14:53:44.375] [main] [info] Extension host with pid 38016 exited with code: 0, signal: null.
[2022-01-26 14:53:44.468] [main] [error] Unable to resolve your shell environment: Unexpected exit code from spawned shell (code null, signal SIGSEGV)
```
some of which might be related.

For comparison, the data from the stable release:
```
Version: 1.63.2 (Universal)
Commit: 899d46d82c4c95423fb7e10e68eba52050e30ba3
Date: 2021-12-15T09:37:28.172Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin x64 21.2.0
```
and `main.log` snippet:
```
[2022-01-26 15:05:15.701] [main] [error] Unable to resolve your shell environment: Unexpected exit code from spawned shell (code null, signal SIGSEGV)
[2022-01-26 15:05:16.267] [main] [info] Starting extension host with pid 41822 (fork() took 4 ms).
[2022-01-26 15:05:16.564] [main] [error] Unable to resolve your shell environment: Unexpected exit code from spawned shell (code null, signal SIGSEGV)
[2022-01-26 15:05:16.749] [main] [error] RequestService#request resolving shell environment failed {}
```

**Comment by gerardkok** [comment]
After having another application segfault, I found out through the crash report, that these crashes are caused by me setting DYLD_LIBRARY_PATH during shell initialisation. Removing that altogether fixes both VSCode and the other application (and also VSCode's exploration build).

The application depending on the setting of DYLD_LIBRARY_PATH (mysql server) also still seems to run fine; apparently this setting was useful in the past, but it is no longer necessary.

I've searched a bit why this started happening after upgrading to Monterey, and my hypothesis is that SIP now handles DYLD_LIBRARY_PATH settings stricter than before. I didn't even know it was 'under SIP', but apparently it is.
